### PR TITLE
enable single gpu training

### PR DESCRIPTION
**Pull request by swdsld**
Before I start, thank you for sharing your awesome work!

The thing is, I tried to train the model with single gpu but I failed since it only support DDP training so I slightly modified your training script to support both single/multi gpu training.

I tested on both single/multi gpu environments. To train with single gpu, execute without DDP launch, and change GPUS to (0, ) in config file.

Thanks.

### CODE REVIEW

File: tools/train_net.py [hunk 1]
```diff
@@ -67,7 +78,7 @@ def main():
     distributed = len(gpus) > 1
     device = torch.device('cuda:{}'.format(args.local_rank))
 
-    if distributed:
+    if distributed is False:
```
Comment (bowenc0221) [inline tools/train_net.py:81]: It this a typo? Why initialize backend when `distributed is False`?

File: tools/train_net.py [hunk 2]
```diff
@@ -134,6 +148,8 @@ def main():
             # data
             start_time = time.time()
             data = next(data_loader_iter)
+            if distributed:
```
Comment (bowenc0221) [inline tools/train_net.py:151]: I think DDP already has the function of sending data to the specific device. Please correct me if I'm wrong. Do we 
 only need to do this when `distributed` is False?

**Comment by bowenc0221** [comment]
@swdsld Thanks for enabling training with a single GPU! Please see my questions above.

File: tools/train_net.py (hunk 1 above)
Comment (swdsld) [reply tools/train_net.py:81]: I found that when ddp is not used, we need to copy the sample from the batch to the gpu manually, and it was just my mistake doing it in a opposite way.

File: tools/train_net.py (hunk 2 above)
Comment (swdsld) [reply tools/train_net.py:151]: that's right! That's why I corrected with """distributed is False"""

**Comment by swdsld** [comment]
Oh, I think I made a mistake. I'm sorry this my first pull request and I'm not used to it.
Let me fix this.

**Comment by swdsld** [comment]
Sorry for the inconvenience.
This is my first pull request and I wasn't familiar with this. Now I've double checked with my available machines and works well!

Thanks.

File: tools/train_net.py [hunk 3]
```diff
@@ -29,6 +29,17 @@
...
+    if type(batch) == torch.Tensor:
+        batch = batch.to(device)
+    elif type(batch) == dict:
+        for key in batch.keys():
+            batch[key] = to_cuda(batch[key], device)
+    elif type(batch) == list:
+        for i in range(len(batch)):
+            batch[i] = to_cuda(batch[i], device)
+    return batch
+
```
Comment (bowenc0221) [inline tools/train_net.py:42]: Could you move this function to `utils` in [utils.py](https://github.com/bowenc0221/panoptic-deeplab/blob/master/segmentation/utils/utils.py)?

File: tools/train_net.py [hunk 4]
```diff
@@ -94,16 +106,18 @@ def main():
     start_iter = 0
     max_iter = config.TRAIN.MAX_ITER
     best_param_group_id = get_lr_group_id(optimizer)
+    
+    get_module = lambda model: model.module if distributed else model
```
Comment (bowenc0221) [inline tools/train_net.py:110]: Same for this. Please write it in a function (not using lambda function) and also put it in [utils.py](https://github.com/bowenc0221/panoptic-deeplab/blob/master/segmentation/utils/utils.py).

File: tools/train_net.py [hunk 5]
```diff
@@ -134,6 +148,8 @@ def main():
             # data
             start_time = time.time()
             data = next(data_loader_iter)
+            if distributed is False:
```
Comment (bowenc0221) [inline tools/train_net.py:151]:
```suggestion
            if not distributed:
```

Please change this.

**Comment by swdsld** [comment]
Thank you for your suggestion. I applied your comments and checked if there's any other conflicts! Sorry there are some messed up commits;;

**Comment by bowenc0221** [comment]
Looks good to me! Thanks for working on this. I'm going to merge this pull request.

**Comment by swdsld** [comment]
Thank you :+1:

**Pull request merged**